libkbfs/folder_branch_ops.go

```diff
@@ -4074,53 +4069,92 @@ func (fbo *folderBranchOps) searchForNode(ctx context.Context,
 	return n, nil
 }
 
-func (fbo *folderBranchOps) unlinkFromCache(op op, unlinkPath path) error {
-	// The entry could be under any one of the unref'd blocks, and
-	// it's safe to perform this when the pointer isn't real, so just
-	// try them all to avoid the overhead of looking up the right
-	// pointer in the old version of the block.
-	for _, ptr := range op.Unrefs() {
-		// It's ok to modify this path, since we break as soon as the
-		// node cache takes a reference to it.
-		unlinkPath.path[len(unlinkPath.path)-1].BlockPointer = ptr
-		found := fbo.nodeCache.Unlink(ptr.Ref(), unlinkPath)
-		if found {
-			break
-		}
-	}
+func (fbo *folderBranchOps) unlinkFromCache(
+	op op, unlinkPath path, unlinkDe DirEntry) error {
```

Comment:
Oops, I think I do that in a later PR, but it's easy enough to do it here, so will do.